query-engine/driver-adapters/src/wasm/result.rs

```diff
@@ -0,0 +1,67 @@
+use js_sys::Boolean as JsBoolean;
+use quaint::error::{Error as QuaintError, ErrorKind};
+use wasm_bindgen::{JsCast, JsValue};
+
+use super::from_js::FromJsValue;
+use crate::{error::DriverAdapterError, JsObjectExtern};
+
+impl From<DriverAdapterError> for QuaintError {
+    fn from(value: DriverAdapterError) -> Self {
+        match value {
+            DriverAdapterError::UnsupportedNativeDataType { native_type } => {
+                QuaintError::builder(ErrorKind::UnsupportedColumnType {
+                    column_type: native_type,
+                })
+                .build()
+            }
+            DriverAdapterError::GenericJs { id } => QuaintError::external_error(id),
+            DriverAdapterError::Postgres(e) => e.into(),
+            DriverAdapterError::Mysql(e) => e.into(),
+            DriverAdapterError::Sqlite(e) => e.into(),
+            // in future, more error types would be added and we'll need to convert them to proper QuaintErrors here
+        }
+    }
+}
+
+/// Wrapper for JS-side result type
+pub(crate) enum JsResult<T>
+where
+    T: FromJsValue,
```

Comment:
TODO: add a comment that this is almost a duplicated of `napi::result::JsResult`, except that generic type bounds can't currently be applied conditionally based on the compilation target.